include/vkwrap/vkwrap/mman.h

```diff
@@ -0,0 +1,556 @@
+            : Base{ createCommandBuffer( device, cmd_pool ) },
+              m_queue{ queue }
+        {
+            begin();
+        } // oneTimeCommand
+
+        void submitAndWait() const
+        {
+            end();
+
+            // clang-format off
+            vk::SubmitInfo submit_info{ 
+                .commandBufferCount = 1, 
+                .pCommandBuffers = &Base::get() };
+            // clang-format on
+
+            m_queue.submit( std::array{ submit_info } );
+            m_queue.waitIdle();
+        } // submitAndWait
+
+        using Base::operator->;
+    }; // class OneTimeCommand
+
+  private:
+    VmaAllocator m_vma;
+    vk::Device m_device;
+    vk::Queue m_queue;
+    vk::CommandPool m_cmd_pool;
+
+    std::unordered_map<vk::Buffer, BufferInfo> m_buffers_info;
+    std::unordered_map<vk::Image, ImageInfo> m_images_info;
+
+    BufferInfo* findInfo( vk::Buffer buffer )
+    {
+        if ( auto entry = m_buffers_info.find( buffer ); entry == m_buffers_info.end() )
+        {
+            throw Error{ "Mman: buffer info not found." };
+        } else
+        {
+            return &entry->second;
+        }
+    } // findInfo
+
+    ImageInfo* findInfo( vk::Image image )
+    {
+        if ( auto entry = m_images_info.find( image ); entry == m_images_info.end() )
+        {
+            throw Error{ "Mman: image info not found." };
+        } else
+        {
+            return &entry->second;
+        }
+    } // findInfo
+
+    VmaAllocation* findAllocation( auto target ) { return &findInfo( target )->allocation; }
+
+    void addInfo( vk::Buffer buffer, const BufferInfo& info ) { m_buffers_info.insert( { buffer, info } ); }
+    void addInfo( vk::Image image, const ImageInfo& info ) { m_images_info.insert( { image, info } ); }
+
+    void clearInfo( vk::Buffer buffer ) { m_buffers_info.erase( buffer ); }
+    void clearInfo( vk::Image image ) { m_images_info.erase( image ); }
+
+    OneTimeCommand createCommand() const { return { m_device, m_cmd_pool, m_queue }; }
+
+    VmaDetailedStatistics getTotalStats() const
+    {
+        VmaTotalStatistics stats{};
+        vmaCalculateStatistics( m_vma, &stats );
+
+        return stats.total;
+    } // getTotalStats
+
+// G++ throws unnecessary warning here.
+#if defined( __GNUG__ )
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wmissing-field-initializers"
+    static constexpr VmaAllocationCreateInfo k_buffer_alloc_create_info{
+        .flags = VMA_ALLOCATION_CREATE_HOST_ACCESS_RANDOM_BIT,
+        .usage = VMA_MEMORY_USAGE_AUTO };
+
+    static constexpr VmaAllocationCreateInfo k_image_alloc_create_info{
+        .usage = VMA_MEMORY_USAGE_AUTO,
+        .requiredFlags = VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT };
+
+    static constexpr VmaVulkanFunctions k_vulkan_functions = {
+        .vkGetInstanceProcAddr = &vkGetInstanceProcAddr,
+        .vkGetDeviceProcAddr = &vkGetDeviceProcAddr };
+#pragma GCC diagnostic pop
+#endif // defined( __GNUG__ )
```

Comment:
```#pragma``` are a code smell. I think ```-Wmissing-field-initializers``` warnings should be disabled for the whole project, because they give way too many false positives

Reply:
Yep, I've removed that option.